packages/site/cms-site/src/iframebridge/IFrameBridge.tsx

```diff
@@ -85,38 +85,47 @@ export const IFrameBridgeProvider = ({ children }: PropsWithChildren) => {
+        const newPreviewElementsData = previewElements
+            .map((previewElement) => {
+                const childNodes = getRecursiveChildrenOfPreviewElement(previewElement.element);
+                const positioning = getCombinedPositioningOfElements(childNodes);
 
-                    return {
-                        adminRoute: previewElement.adminRoute,
-                        label: previewElement.label,
-                        position: {
-                            top: positioning.top,
-                            left: positioning.left,
-                            width: positioning.width,
-                            height: positioning.height,
-                        },
-                    };
-                })
-                .sort((previousElementData, nextElementData) => {
-                    const previousSize = previousElementData.position.width * previousElementData.position.height;
-                    const nextSize = nextElementData.position.width * nextElementData.position.height;
-                    return nextSize - previousSize;
-                })
-                .map((elementData, index) => {
-                    return {
-                        ...elementData,
-                        position: {
-                            ...elementData.position,
-                            zIndex: index + 1,
-                        },
-                    };
-                }),
-        );
+                return {
+                    adminRoute: previewElement.adminRoute,
+                    label: previewElement.label,
+                    position: {
+                        top: positioning.top,
+                        left: positioning.left,
+                        width: positioning.width,
+                        height: positioning.height,
+                    },
+                };
+            })
+            .sort((previousElementData, nextElementData) => {
+                const previousSize = previousElementData.position.width * previousElementData.position.height;
+                const nextSize = nextElementData.position.width * nextElementData.position.height;
+                return nextSize - previousSize;
+            })
+            .map((elementData, index) => {
+                return {
+                    ...elementData,
+                    position: {
+                        ...elementData.position,
+                        zIndex: index + 1,
+                    },
+                };
+            });
+
+        setPreviewElementsData((previousElementsData) => {
+            const dataDidNotChange = JSON.stringify(previousElementsData) === JSON.stringify(newPreviewElementsData);
+
+            if (dataDidNotChange) {
+                // Returning the previous object (same reference) prevents the state-update from triggering a re-render
+                return previousElementsData;
+            }
+
+            return newPreviewElementsData;
+        });
```

Comment:
Is this the best solution to this problem? Wouldn't it work to set [attributesFilter](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe#attributefilter) in the mutation observer and only listen for `class` (and maybe `style`)?

Reply:
The other idea i had to fix this issue specifically, was to ignore the mutation, when only the `name` and `type` of an input element would change: https://github.com/vivid-planet/comet/pull/3093

However it does not make sense in general to update and re-render the preview-data when there is no change in the preview-data. Since this is a logical thing to do anyway and also fixes the issue I decided this to be the better of the two fixes.

Reply:
Also using `attributesFilter` could in theory cause new issues, e.g. if the size/position of the element depends on something other than simply `class` or `style` like a data-attribute that influences the styling.